### PR TITLE
Enable travis-ci and coverage.

### PR DESCRIPTION
**Pull request by bjlittle**
@JohannesBuchner Okay, so I've separated my concerns within PR #33 (as you requested) and reduced the scope down to a simple refactor of the testing, enabling travis-ci and including automated code coverage for testing.

The changes to the `setup.py` are required to interface with travis-ci i.e. `python setup.py test`.

I've also replaced the stock testing image, which for very different reasons is still lovely.

If you're happy with this PR I'd prefer that you first enabled [travis-ci](https://travis-ci.org) for this (i.e. your) repo, as well as [coveralls](https://coveralls.io). Let me know when you have done this and I'll re-push the PR to trigger the web hooks so that we can see the tests run on travis and the coverage report .... then you can merge, if you're happy.

View the reports for this PR on my branch:
* https://travis-ci.org/bjlittle/imagehash/builds/171509096
* https://coveralls.io/github/bjlittle/imagehash

### CODE REVIEW

**Comment by bjlittle** [comment]
@JohannesBuchner The interesting kicker for this PR is that Python3 is forcing your hand with regards to pep8 :wink: 

See [travis-ci Python 3.4](https://travis-ci.org/bjlittle/imagehash/jobs/171509098) and [travis-ci Python3.5](https://travis-ci.org/bjlittle/imagehash/jobs/171509099)

That is, the formatting in your code is non-compliant.

My suggestion, as a plan of action, is that if you're happy with this PR, merge it as is (including any follow-up actions that you may request) **but** with the travis-ci tests failing for Python3.

My next PR will fix this. The way to do this is to include my pep8 changes from #33 and include the pep8 testing to ensure pep8 coding compliance is enforced. 

If you want `imagehash` working in Python3, then that's the change we'll need to make ... your call @JohannesBuchner

File: imagehash/tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import os.path
+import unittest
+
+from PIL import Image
+
+import imagehash
+
+
+class TestImageHash(unittest.TestCase):
+    def get_data_image(self, fname=None):
```
Comment (bjlittle) [inline imagehash/tests/__init__.py]: @JohannesBuchner On reflection, this could actually be a `@staticmethod`

**Comment by JohannesBuchner** [comment]
Please merge commit 4d93715095 which fixes formatting issues introduced in the update-hex-to-hash branch. I will merge this PR over the weekend.

**Comment by bjlittle** [comment]
@JohannesBuchner I've rebased this PR.

Could you please enable travis-ci (and coverage) on this repo before merging so that we can see the tests running successfully for this PR on the master branch. Thanks.

**Comment by JohannesBuchner** [comment]
I activated travis-ci and coverall for this repo.

File: setup.py [hunk 2]
```diff
@@ -1,18 +1,28 @@
...
+
+
+setup_args = dict(
     name='ImageHash',
     version='3.0',
     author='Johannes Buchner',
     author_email='buchner.johannes@gmx.at',
-    packages=['imagehash'],
+    packages=['imagehash', 'imagehash.tests'],
+    package_data=package_data,
```
Comment (JohannesBuchner) [inline setup.py]: Could you please change this to 
`package_data=dict(imagehash='tests/data/*`,

File: setup.py [hunk 3]
```diff
@@ -1,18 +1,28 @@
+import os
+import os.path
+
 try:
     from setuptools import setup
 except ImportError:
     from distutils.core import setup
 
+
```
Comment (JohannesBuchner) [inline setup.py]: Please remove

File: setup.py [hunk 4]
```diff
@@ -24,5 +34,9 @@
         "pillow",      # or PIL
         "PyWavelets",  # for whash
     ],
+    test_suite='imagehash.tests'
 )
 
+
+if __name__ == '__main__':
```
Comment (JohannesBuchner) [inline setup.py]: Why is this needed? Because python setup.py test does not work otherwise?

File: setup.py (hunk 3 above)
Comment (bjlittle) [inline setup.py]: Okay

File: setup.py (hunk 4 above)
Comment (bjlittle) [inline setup.py]: Basically, previously `setuptools.setup` was being called at the module level. Which is fine. However, the disadvantage of this is that it **always** executes, even on `import` which you may not want to happen, particularly if a developer is changing `setup.py` and wants to see changes to the contents of the dictionary passed into `setuptools.setup`. Which is normal practice in several projects that I've seen.

I'm going to remove it, as it's clear to me that you want minimal changes. You've now seen this pattern and if you want it, you can easily re-implement it.

**Comment by bjlittle** [comment]
@JohannesBuchner Looks good to me ... :smile:

**Comment by JohannesBuchner** [comment]
That looks great, thank you very much for these changes!

**Comment by JohannesBuchner** [comment]
version 3.1 is on pypi for you

**Comment by bjlittle** [comment]
@JohannesBuchner Awesome, that's really great news!
